### PR TITLE
AP_CANManager: don't drop slcan passthrough packet when CAN send fails

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

File: libraries/AP_CANManager/AP_SLCANIface.cpp [hunk 1]
```diff
@@ -685,7 +685,13 @@ int16_t SLCAN::CANIface::receive(AP_HAL::CANFrame& out_frame, uint64_t& rx_time,
         // Also send this frame over can_iface when in passthrough mode,
         // We just push this frame without caring for priority etc
         if (_can_iface) {
-            _can_iface->send(out_frame, AP_HAL::native_micros64() + 1000, out_flags);
+            if (_can_iface->send(out_frame, AP_HAL::native_micros64() + 1000, out_flags)) {
+                rx_queue_.pop();
```
Comment (tridge) [inline libraries/AP_CANManager/AP_SLCANIface.cpp:689]: should we timeout packets after 8 tries like we do in AP_Periph?

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_CANManager/AP_SLCANIface.cpp (hunk 1 above)
Comment (bugobliterator) [reply libraries/AP_CANManager/AP_SLCANIface.cpp:689]: yeah, that might be a good idea.

**Comment by CUAVcaijie** [comment]
> test environment：
 Mission Planner 1.3.74

> Test hardware:
CUAVv5
CUAV CAN GPS
CUAVv5Nano

> Hardware wiring:
1：Update the firmware of CAN GPS
2：Connect the flight controller used for UAVCAN GUI Tool debugging
3：Update the flight control of CAN GPS ，Connect MP
image

> Test process：
1:CUAVv5Nano connect UAVCAN GUI Tool （CUAVv5Nano 连接到UAVCAN GUI Tool）
2:CUAVv5 connect Mission Planner 1.3.74 UAVCAN（CUAVv5 连接到Mission Planner 1.3.74 UAVCAN)

> Test Results:

The file does not have a complete package when updating

image

**Review by tridge — APPROVED** [review]

**Comment by CUAVcaijie** [comment]
CUAV CAN GPS update firmware is normal, check data without packet loss
image
image
image

**Comment by bugobliterator** [comment]
@tridge this is ready to merge

**Comment by Gone4Dirt** [comment]
I have also tested this and it fixes flashing via SLCAN

**Review by tridge — APPROVED** [review]

**Pull request merged**